### DISCUSSION TITLE
Focus not set on 'first control' after callable action

### DISCUSSION THREAD

**Discussion by JoelHaslam** [grafana/grafana#39857] [General]
Easy and fast navigation with only a few keystrokes is essential for keyboard only operation (if users have motor impairments) and making this more efficient is a great way to achieve this.

When selecting an element from the UI, it helps out greatly if the focus sets directly on the first control of the function that the element selected provides to allow the user to interact with it straight away.

**Steps to reproduce**:
From the homepage of a dashboard, tab through to the _Explore_ icon on the left hand side and select the element.

Using the keyboard, try to get focus on the page that has appeared by selecting _Explore_.

**Actual Result**:
You have to continue to cycle through all the other surrounding elements before being able to get focus of the content that's opened when selecting _Explore_

**Expected Result**:
Focus would set directly on the first control of the _Explore_ page (ie the first option available for a query).

**Relevant WCAG Criteria:** [2.4.1 bypass-block](https://www.w3.org/TR/WCAG21/#bypass-blocks) + [How to meet 2.4.1]()https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=241#bypass-blocks) + [Understanding bypass blocks](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks.html)

**Environment**:
- Grafana version: 8.2.0-beta1
- Data source type & version: Default data used at [play.grafana.org](https://play.grafana.org/d/000000012/grafana-play-home?orgId=1)
- User OS & Browser: Windows & Chrome
- Others:

**Comment by JoelHaslam** [comment]
I'd like to drum up some conversation of this discussion item.

Do other think that this impactful to users? Constantly having to reset the keyboard focus every time you move to a new page can start to really impact a user, and having it automatically set to the most useful location would be super helpful!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
